force-app/main/default/classes/newTaskTraining.cls

```diff
@@ -0,0 +1,18 @@
+public class newTaskTraining {
+    public static void createTask() {
+        Date today = Date.today();
+        List<Training__c> endedTrains = [SELECT Name, End_Date__c, Training_Course__r.OwnerId
+                                        FROM Training__c
+                                        WHERE DAY_ONLY(End_Date__c) = :today];
```

Comment:
You can use End_Date__c = TODAY

https://developer.salesforce.com/docs/atlas.en-us.soql_sosl.meta/soql_sosl/sforce_api_calls_soql_select_dateformats.htm